source/_integrations/homekit_controller.markdown

```diff
@@ -45,67 +45,80 @@ ha_integration_type: integration
+There are different methods to add a HomeKit device to Home Assistant:
 
-## Supported devices
+- [via Ethernet or Wi-Fi](#adding-a-homekit-device-via-ethernet-or-wi-fi): Requires the pairing code
+- [via Bluetooth](#adding-a-homekit-device-through-bluetooth): Requires the pairing code
+- [onto a Thread network via Home Assistant](#adding-a-homekit-device-to-a-thread-network-via-home-assistant): No pairing code required
+- [onto a Thread network via Apple Thread border router](#adding-a-homekit-device-via-apple-thread-border-router): No pairing code required
 
-There is currently support for the following device types within Home Assistant:
+## Adding a HomeKit device via Ethernet or Wi-Fi
 
-- Alarm Control Panel (HomeKit security system)
-- Climate (HomeKit thermostats and air conditioners)
-- Cover (HomeKit garage door openers, windows, or window coverings)
-- Light (HomeKit lights)
-- Lock (HomeKit lock)
-- Switch (HomeKit switches, outlets and valves)
-- Binary Sensor (HomeKit motion, contact, occupancy, carbon monoxide and smoke sensors)
-- Sensor (HomeKit humidity, temperature, co2 and light level sensors)
-- Fan
-- Air Quality
-- Humidifier (HomeKit humidifiers and dehumidifiers)
-- Automation Triggers (HomeKit 'stateless' accessories like buttons, remotes and doorbells)
+The HomeKit controller integration automatically detects HomeKit [compatible devices](#supported-devices) that are ready to pair if the [`zeroconf`](/integrations/zeroconf/) integration is enabled. The `zeroconf` is enabled by default on new installations via the [`default_config`](/integrations/default_config/) integration.
 
-<div class='note'>
+### Prerequisites
 
-  If your device is not on this list then you may still be able to pair it, and it will be visible in the Device Registry. But Home Assistant may not create entities for it.
+1. If you do not have the [`default_config`](/integrations/default_config/) integration, add [`zeroconf`] to your configuration file.
+1. Find your HomeKit pairing code. The code is on the device itself, or on the packaging. If your device has a screen, it may be shown on screen.
+   - There is no way to recover this if you do not have it. In this case, you will need to contact the manufacturer to see what options you have.
+2. Power up your device.
+3. If your device is in your network but is paired with an Apple device via HomeKit, reset the device.
+   - Otherwise you won't be able to pair it with Home Assistant.
+   - HomeKit devices can only be paired to a single controller at once.
+4. If your device is not already in your network, join the device to your network:
+   - Follow the instructions of the manufacturer on how to join the device to the network
+   - If you do not have the instructions, or the device is HomeKit only:
+     - Pair it with the Apple Home app (using the iOS Home app).
```

Comment:
So that will help to get it onto the (WiFi) network? I guess this needs unpairing it again after that?

Reply:
Nevermind, I see that this is step one below.

I think I'd move the step below to this prerequisit here, so that the onboarding flow is the same for all devices.